### PR TITLE
Rename sha3* related functions and parameters to keccak* 

### PR DESCRIPTION
**Pull request by JianM95**
**Mainly modified libdevcrypto/hash.h libdevcrypto/hash.cpp and other references. Changed the use of sha3 method names and parameter names to keccak.**

### CODE REVIEW

File: libethcore/TransactionReceipt.h [hunk 1]
```diff
@@ -72,13 +72,13 @@ class TransactionReceipt
         return m_receipt;
     }
 
-    const bytes& sha3()
+    const bytes& keccak()
```
Comment (cyjseagull) [inline libethcore/TransactionReceipt.h:75]: Please modify from `sha3` to `hash`

File: libethcore/Transaction.h [hunk 2]
```diff
@@ -211,8 +211,8 @@ class Transaction
         return out;
     }
 
-    /// @returns the SHA3 hash of the RLP serialisation of this transaction.
-    h256 sha3(IncludeSignature _sig = WithSignature) const;
+    /// @returns the Keccak256 hash of the RLP serialisation of this transaction.
+    h256 keccak256(IncludeSignature _sig = WithSignature) const;
```
Comment (cyjseagull) [inline libethcore/Transaction.h:215]: Please modify from `sha3` to `hash`

File: libdevcrypto/SM3Hash.h [hunk 3]
```diff
@@ -27,7 +27,7 @@
 
 namespace dev
 {
-// SHA-3 convenience routines.
+// Keccak256 convenience routines.
```
Comment (cyjseagull) [inline libdevcrypto/SM3Hash.h:30]: Delete this.

File: libethcore/Block.cpp [hunk 4]
```diff
@@ -243,7 +243,7 @@ std::shared_ptr<std::map<std::string, std::vector<std::string>>> Block::getTrans
     return merklePath;
 }
 
-void Block::getReceiptAndSha3(RLPStream& txReceipts, std::vector<dev::bytes>& receiptList) const
+void Block::getReceiptAndKeccak(RLPStream& txReceipts, std::vector<dev::bytes>& receiptList) const
```
Comment (cyjseagull) [inline libethcore/Block.cpp:246]: Pls modify from `getReceiptAndKeccak` to `getReceiptAndHash`

File: libethcore/BlockHeader.h [hunk 5]
```diff
@@ -290,7 +290,7 @@ class BlockHeader
     h512s m_sealerList;          /// sealer list
     /// -------structure of block header end------
 
-    mutable h256 m_hash;       ///< (Memoised) SHA3 hash of the block header with seal.
+    mutable h256 m_hash;       ///< (Memoised) Keccak256 hash of the block header with seal.
```
Comment (cyjseagull) [inline libethcore/BlockHeader.h:293]: modify to `(Memoised) hash of the block header with seal.`

File: libethcore/TransactionReceipt.h [hunk 6]
```diff
@@ -106,7 +106,7 @@ class TransactionReceipt
 
 private:
     bytes m_receipt = bytes();
-    bytes m_sha3 = bytes();
+    bytes m_keccak = bytes();
```
Comment (cyjseagull) [inline libethcore/TransactionReceipt.h:109]: Pls rename to `m_hash`

File: libethcore/TxsParallelParser.cpp [hunk 7]
```diff
@@ -157,12 +157,12 @@ void TxsParallelParser::decode(std::shared_ptr<Transactions> _txs, bytesConstRef
                         (*_txs)[i]->decode(txBytes.cropped(offset, size), _checkSig);
                         if (_withHash)
                         {
-                            // cache the sha3
-                            // Note: can't calculate sha3 with sha3(txBytes.cropped(offset, size))
+                            // cache the keccak256
```
Comment (cyjseagull) [inline libethcore/TxsParallelParser.cpp:160]: Pls rename `keccak256` to `hash`

File: libinterpreter/VMConfig.h [hunk 8]
```diff
@@ -190,7 +190,7 @@ namespace eth
         &&SAR,                                  \
         &&INVALID,                              \
         &&INVALID,                              \
-        &&SHA3, /* 20, */                       \
```
Comment (bxq2011hust) [inline libinterpreter/VMConfig.h:193]: don't modify all files in libinterpreter dir

**Review by cyjseagull — APPROVED** [review]

**Review by bxq2011hust — APPROVED** [review]

**Pull request merged**